### ISSUE TITLE
Use Safari View Controller instead of redirecting traffic to the Safari app

### ISSUE THREAD

**Issue by iC0rraxX** [damus-io/damus#295]
Currently, when a user clicks on a link, they are redirected to the iPhone's native browser app, Safari. However, using the Safari View Controller, you can get all the benefits of Safari within the app without building an in-app browser from scratch. Usually, 10-15 lines of code are enough to implement, so I think it can be a great addition without needing too much commitment.

**Comment by olheadinternet** [comment]
just my two sats, but i hate having to go through the extra clicks when i do want to open a link in safari proper. Often i want to look at the content that was shared with the link, whilst looking at the link, webview in-app prevents this. This is one of those retention tactics to keep you in the app so you go back to looking at their ads, not needed in a nostr client not based on advertising.

**Comment by iC0rraxX** [comment]
> just my two sats, but i hate having to go through the extra clicks when i do want to open a link in safari proper. Often i want to look at the content that was shared with the link, whilst looking at the link, webview in-app prevents this. This is one of those retention tactics to keep you in the app so you go back to looking at their ads, not needed in a nostr client not based on advertising.



I agree that normally this is a retention tactic, but for me, it is actually a good addition. I hate wasting time going back and forth between apps, especially on Nostr where I open links all the time. If you implement SVC, however, why would a user go to Safari? It loads cookies, autofill passwords, and any browser extensions directly on the page. Anyway thanks for sharing, I didn't even think it would bother some users. At this point, I think the best way is to have the user decide it via a flag in the settings.
